Base/Python/slicer/util.py

```diff
@@ -2497,6 +2501,148 @@ def dataframeFromMarkups(markupsNode):
+
+    By default, voxels values are deep-copied, therefore if the ITK image is modified
+    after calling this method, voxel values in the volume node will not change.
+
+    See :py:meth:`updateVolumeFromITKImage` to understand memory ownership.
+
+    :param itkImage: ITK image containing volume voxels.
+    :param name: volume node name
+    :param nodeClassName: type of created volume, default: ``vtkMRMLScalarVolumeNode``.
+      Use ``vtkMRMLLabelMapVolumeNode`` for labelmap volume, ``vtkMRMLVectorVolumeNode`` for vector volume.
+    :param deepCopy: Whether voxels values are deep-copied or not.
+
+    :return: created new volume node
+    """
+    import slicer
+
+    if name is None:
+        name = ""
+    if nodeClassName is None:
+        nodeClassName = "vtkMRMLScalarVolumeNode"
+
+    volumeNode = slicer.mrmlScene.AddNewNodeByClass(nodeClassName, name)
+    updateVolumeFromITKImage(volumeNode, itkImage, deepCopy)
+    volumeNode.CreateDefaultDisplayNodes()
+
+    return volumeNode
+
+
+def updateVolumeFromITKImage(volumeNode, itkImage, deepCopy=True):
+    """Set voxels of a volume node from an ITK image.
+
+    By default, voxels values are deep-copied, therefore if the ITK image is modified
+    after calling this method, voxel values in the volume node will not change.
+
+    .. warning:: Important: Setting `deepCopy` to False means that the memory area is
+      shared between the ITK image and the ``vtkImageData`` object in the MRML volume node,
+      therefore modifying the ITK image values requires to call :py:meth:`itkImageFromVolumeModified`.
+
+      If the ITK image is reallocated, calling this function is required.
+    """
+    import itk
+    import vtk
+    import vtkAddon
+
+    # Convert ITK image to VTK image
+    vtkImage = itk.vtk_image_from_image(itkImage)
+
+    # Get direction from VTK image
+    ijkToRASMatrix = vtk.vtkMatrix4x4()
+    vtkAddon.vtkAddonMathUtilities.SetOrientationMatrix(vtkImage.GetDirectionMatrix(), ijkToRASMatrix)
+
+    # Transform from LPS (used in ITK) to RAS (used in Slicer)
+    lpsToRas = vtk.vtkMatrix4x4()
+    lpsToRas.SetElement(0, 0, -1)
+    lpsToRas.SetElement(1, 1, -1)
+    vtk.vtkMatrix4x4.Multiply4x4(lpsToRas, ijkToRASMatrix, ijkToRASMatrix)
+
+    # Update output node direction/origin/spacing
+    volumeNode.SetIJKToRASMatrix(ijkToRASMatrix)
+    volumeNode.SetOrigin(*vtkImage.GetOrigin())
+    volumeNode.SetSpacing(*vtkImage.GetSpacing())
+
+    # Reset direction/origin/spacing of the VTK output image.
+    # See https://github.com/Slicer/Slicer/issues/6911
+    identidyMatrix = vtk.vtkMatrix3x3()
+    vtkImage.SetDirectionMatrix(identidyMatrix)
+    vtkImage.SetOrigin((0, 0, 0))
+    vtkImage.SetSpacing((1., 1., 1.))
+
+    # Update output node setting VTK image data
+    if deepCopy:
+        vtkImageCopy = vtk.vtkImageData()
+        vtkImageCopy.DeepCopy(vtkImage)
+        volumeNode.SetAndObserveImageData(vtkImageCopy)
```

Comment:
@lassoan `deepCopy` is now the default

@dzenanz I tested the changes in the context of SlicerITKUltrasound and it didn't see any issues (with/without speechifying deepCopy to `False`)